src/sonic-yang-models/yang-models/sonic-syslog.yang

```diff
@@ -9,22 +9,72 @@ module sonic-syslog {
         prefix inet;
     }
 
+    import sonic-mgmt_vrf {
+        prefix mvrf;
+    }
+
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    description "Syslog YANG Module for SONiC OS: remote syslog logging";
+
+    revision 2022-04-18 {
+        description "Initial revision.";
+    }
+
+    typedef vrf-device {
+        description "Represents syslog VRF device";
+        type enumeration {
+            enum default;
+            enum mgmt;
+        }
+    }
+
     container sonic-syslog {
 
         container SYSLOG_SERVER {
 
-            description "SYSLOG SERVER part of config_db.json";
+            description "SYSLOG_SERVER part of config_db.json";
 
             list SYSLOG_SERVER_LIST {
 
                 key "server_address";
 
                 leaf server_address {
-                    description "syslog server ip address";
+                    description "Syslog server IP address";
+                    type inet:ip-address;
+                }
+
+                leaf source {
+                    description "Syslog source IP address";
                     type inet:ip-address;
+                    must "(contains(current(), '.') and contains(../server_address, '.'))
```

Comment:
>. [](http://example.com/codeflow?start=48&length=1)

ipv6 address could also container `.`. Please check [https://en.wikipedia.org/w/index.php?title=IPv6_address&section=19#Special_addresses](https://en.wikipedia.org/w/index.php?title=IPv6_address&section=19#Special_addresses)

Reply:
@qiluo-msft do we have such a use case in SONiC?